common/trajectory.cpp

```diff
@@ -492,15 +492,20 @@ Trajectory Trajectory::rescaledToVelocity(
 Trajectory Trajectory::createWilliamsonTurn(
 		double turnRadius,
 		double acceleration,
+		double minSpeed,
+		double maxSpeed,
 		TrajectoryPoint startPoint,
 		std::chrono::milliseconds startTime)
 {
 
 	using namespace std::chrono;
 	using Eigen::MatrixXd;
 
-	constexpr double topSpeed = 10 / 3.6;
-	const int calculatedNoOfPoints = 500;
+	double topSpeed = turnRadius*2 / 3.6; // Set top speed depending on radius.
+	// Limit top speed.
+	std::clamp(topSpeed, minSpeed, maxSpeed);
+
+	const int calculatedNoOfPoints = 200;
```

Comment:
Can this hard-coding lead to problems?

Reply:
Not sure, it has always been hardcoded, but maybe the number of points should be proportional to the radius, something like turnRadius*100 might be suitable

Reply:
Made a quick investigation and it seems like it needs to be a constant integer for the matrices created later on to be correct, which basically means that the variable needs to be hardcoded

Reply:
@victorjarlow  Made an attempt to implement dynamic variable and matrices but it wasn't really possible, primarily the "eigen::array"s were difficult with the rest of the code. I think if you still want the dynamic variable, we can write it as an issue and handle it separately.

Reply:
```suggestion
	const int calculatedNoOfPoints = 200; // TODO: Make this variable dynamic depending on turnRadius
```